.github/workflows/golden_tests.yml

```
@@ -0,0 +1,101 @@
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  golden_tests:
+    name: Golden Tests
+    runs-on: ubuntu-latest
+    continue-on-error: true # Don't fail the build if golden tests fail
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)